### PR TITLE
Refactored metric tagging

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Description

Adding metrics tags has now been refactored, similar to span decorators.

## Motivation and Context

- metrics tags are now aligned with tracing tags
- retry metrics have been changed from dedicated metrics to new tag (`retry_number`) on existing `http.client.requests` metrics
- tag generators have been extended and split up to simplify composition
- static tags can now be set per `Http` instance
- service loader support for tag generation

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-micrometer/src/main/java/org/zalando/riptide/micrometer/tag/TagGenerator.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.google.common.collect.Lists;
+import io.micrometer.core.instrument.Tag;
+import org.apiguardian.api.API;
+import org.springframework.http.client.ClientHttpResponse;
+import org.zalando.riptide.RequestArguments;
+
+import javax.annotation.CheckReturnValue;
+import java.io.IOException;
+import java.util.Arrays;
```
Comment (whiskeysierra) [inline riptide-micrometer/src/main/java/org/zalando/riptide/micrometer/tag/TagGenerator.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Arrays'](https://app.codacy.com/app/whiskeysierra/riptide/pullRequest?prid=4483828)

**Comment by fatroom** [comment]
👍

File: riptide-micrometer/src/main/java/org/zalando/riptide/micrometer/tag/TagGenerator.java (hunk 1 above)
Comment (whiskeysierra) [inline riptide-micrometer/src/main/java/org/zalando/riptide/micrometer/tag/TagGenerator.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Arrays'](https://app.codacy.com/app/whiskeysierra/riptide/pullRequest?prid=4483828)

Comment (whiskeysierra) [inline riptide-micrometer/src/main/java/org/zalando/riptide/micrometer/tag/TagGenerator.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Arrays'](https://app.codacy.com/app/whiskeysierra/riptide/pullRequest?prid=4483828)

**Comment by whiskeysierra** [comment]
@fatroom Can you give me another :+1:?

**Comment by fatroom** [comment]
👍

**Pull request merged**